kong/router/fields.lua

```diff
@@ -209,7 +218,67 @@ if is_http then
 
         return params.queries[field:sub(PREFIX_LEN + 1)]
       end
-    end
+
+    elseif field:sub(1, HTTP_SEGMENTS_PREFIX_LEN) == HTTP_SEGMENTS_PREFIX then
+      return function(params)
+        if not params.segments then
+          HTTP_SEGMENTS_REG_CTX.pos = 2 -- reset ctx, skip first '/'
+          params.segments = re_split(params.uri, "/", "jo", HTTP_SEGMENTS_REG_CTX)
+        end
+
+        local segments = params.segments
+
+        local name = field:sub(HTTP_SEGMENTS_PREFIX_LEN + 1)
```

Comment:
```suggestion
        local range = field:sub(HTTP_SEGMENTS_PREFIX_LEN + 1)
```

is better than "name".